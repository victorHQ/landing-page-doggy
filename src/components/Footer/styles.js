import styled, { css } from 'styled-components';
import { Container as SectionContainer } from '../SectionContainer/styles';

export const Container = styled.footer`
  ${({ theme }) => css`
    text-align: center;
    border-top: 0.1rem solid ${theme.colors.mediumGray};

    ${SectionContainer} {
      padding: ${theme.spacings.xsmall};
    }

    a {
      color: inherit;
      text-decoration: none;
      font-size: ${theme.font.sizes.small};
    }

    p{
      font-size: ${theme.font.sizes.small};
      font-weight: bold;
    }
  `}
`;
