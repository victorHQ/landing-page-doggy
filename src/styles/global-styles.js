import { createGlobalStyle } from 'styled-components';

import OpenSansV29LatinRegularWoff from '../fonts/open-sans-v29-latin/open-sans-v29-latin-regular.woff';
import OpenSansV29LatinRegularWoff2 from '../fonts/open-sans-v29-latin/open-sans-v29-latin-regular.woff2';
import OpenSansV29LatinItalicWoff from '../fonts/open-sans-v29-latin/open-sans-v29-latin-italic.woff';
import OpenSansV29LatinItalicWoff2 from '../fonts/open-sans-v29-latin/open-sans-v29-latin-italic.woff2';
import OpenSansV29Latin700Woff from '../fonts/open-sans-v29-latin/open-sans-v29-latin-700.woff';
import OpenSansV29Latin700Woff2 from '../fonts/open-sans-v29-latin/open-sans-v29-latin-700.woff2';
import OpenSansV29Latin700ItalicWoff from '../fonts/open-sans-v29-latin/open-sans-v29-latin-700italic.woff';
import OpenSansV29Latin700ItalicWoff2 from '../fonts/open-sans-v29-latin/open-sans-v29-latin-700italic.woff2';
import MontserratV24Latin900Woff from '../fonts/montserrat-v24-latin/montserrat-v24-latin-900.woff';
import MontserratV24Latin900Woff2 from '../fonts/montserrat-v24-latin/montserrat-v24-latin-900.woff2';
import MontserratV24Latin900ItalicWoff from '../fonts/montserrat-v24-latin/montserrat-v24-latin-900italic.woff';
import MontserratV24Latin900ItalicWoff2 from '../fonts/montserrat-v24-latin/montserrat-v24-latin-900italic.woff2';

export const GlobalStyles = createGlobalStyle`
  @font-face {
    font-family: 'Open Sans';
    font-style: normal;
    font-weight: 400;
    src: local(''),
        url(${OpenSansV29LatinRegularWoff2}) format('woff2'), /* Chrome 26+, Opera 23+, Firefox 39+ */
        url(${OpenSansV29LatinRegularWoff}) format('woff'); /* Chrome 6+, Firefox 3.6+, IE 9+, Safari 5.1+ */
  }

  @font-face {
    font-family: 'Open Sans';
    font-style: italic;
    font-weight: 400;
    src: local(''),
        url(${OpenSansV29LatinItalicWoff2}) format('woff2'), /* Chrome 26+, Opera 23+, Firefox 39+ */
        url(${OpenSansV29LatinItalicWoff}) format('woff'); /* Chrome 6+, Firefox 3.6+, IE 9+, Safari 5.1+ */
  }

  @font-face {
    font-family: 'Open Sans';
    font-style: normal;
    font-weight: 700;
    src: local(''),
        url(${OpenSansV29Latin700Woff2}) format('woff2'), /* Chrome 26+, Opera 23+, Firefox 39+ */
        url(${OpenSansV29Latin700Woff}) format('woff'); /* Chrome 6+, Firefox 3.6+, IE 9+, Safari 5.1+ */
  }
  @font-face {
    font-family: 'Open Sans';
    font-style: italic;
    font-weight: 700;
    src: local(''),
        url(${OpenSansV29Latin700ItalicWoff2}) format('woff2'), /* Chrome 26+, Opera 23+, Firefox 39+ */
        url(${OpenSansV29Latin700ItalicWoff}) format('woff'); /* Chrome 6+, Firefox 3.6+, IE 9+, Safari 5.1+ */
  }

  @font-face {
    font-family: 'Montserrat';
    font-style: normal;
    font-weight: 900;
    src: local(''),
        url(${MontserratV24Latin900Woff2}) format('woff2'), /* Chrome 26+, Opera 23+, Firefox 39+ */
        url(${MontserratV24Latin900Woff}) format('woff'); /* Chrome 6+, Firefox 3.6+, IE 9+, Safari 5.1+ */
  }
  /* montserrat-900italic - latin */
  @font-face {
    font-family: 'Montserrat';
    font-style: italic;
    font-weight: 900;
    src: local(''),
        url(${MontserratV24Latin900ItalicWoff2}) format('woff2'), /* Chrome 26+, Opera 23+, Firefox 39+ */
        url(${MontserratV24Latin900ItalicWoff}) format('woff'); /* Chrome 6+, Firefox 3.6+, IE 9+, Safari 5.1+ */
  }

  * {
    margin: 0;
    padding: 0;
    box-sizing: border-box;
  }

  html {
    font-size: 62.5%;
    scroll-behavior: smooth;
  }

  body {
    font-size: 1.6rem;
    font-family: ${({ theme }) => theme.font.family.default};
  }

  h1, h2, h3, h4, h5, h6 {
    font-family: ${({ theme }) => theme.font.family.secondary};
    margin: ${({ theme }) => theme.spacings.large} 0;
  }

  p {
    margin: ${({ theme }) => theme.spacings.medium} 0;
  }

  ul, ol {
    margin: ${({ theme }) => theme.spacings.medium};
    padding: ${({ theme }) => theme.spacings.medium};
  }

  a {
    color: ${({ theme }) => theme.colors.secondaryColor};
  }
`;
