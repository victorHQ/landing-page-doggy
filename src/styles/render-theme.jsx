import { render } from '@testing-library/react';
import { ThemeProvider } from 'styled-components';
import { theme } from './theme';
import PropTypes from 'prop-types';

const Theme = ({ children }) => {
  return <ThemeProvider theme={theme}>{children}</ThemeProvider>;
};

Theme.propTypes = {
  children: PropTypes.node.isRequired,
};

export const renderTheme = (children) => {
  const rendered = render(<Theme>{children}</Theme>);

  const { rerender } = rendered;
  rendered.rerender = (children) => rerender(<Theme>{children}</Theme>);

  return rendered;
};
