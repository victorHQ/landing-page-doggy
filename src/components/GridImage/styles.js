import styled, { css } from 'styled-components';
import { Title as HeadingContainer } from '../Heading/styles';
import { Container as TextComponent } from '../TextComponent/styles';

export const Container = styled.div`
  ${({ theme }) => css`
    > ${TextComponent} {
      margin-bottom: ${theme.spacings.xhuge};
    }
  `}
`;

export const Grid = styled.div`
  ${({ theme }) => css`
    display: grid;
    grid-template-columns: repeat(3, 1fr);
    gap: ${theme.spacings.large};

    @media ${theme.media.lteMedium} {
      grid-template-columns: 1fr;
    }
  `}
`;

export const GridElement = styled.div`
  ${({ theme }) => css`
    max-width: 35rem;
    max-height: 35rem;
    overflow: hidden;
  `}
`;

export const Image = styled.img`
  ${({ theme }) => css`
    width: 100%;
    transition: all 300ms ease-in-out;

    &:hover {
      transform: scale(1.2) rotate(10deg);
    }
  `}
`;
