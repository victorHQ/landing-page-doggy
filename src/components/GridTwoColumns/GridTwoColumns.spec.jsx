import { renderTheme } from '../../styles/render-theme';
import { GridTwoColumns } from '.';

import mock from './mock';

describe('<GridTwoColum />', () => {
  it('should render two column grid', () => {
    const { container } = renderTheme(<GridTwoColumns {...mock} />);
    expect(container).toMatchSnapshot();
  });
});
