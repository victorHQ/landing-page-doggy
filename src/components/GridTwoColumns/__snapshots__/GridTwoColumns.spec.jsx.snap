// Vitest Snapshot v1

exports[`<GridTwoColum /> > should render two column grid 1`] = `
.c5 {
  color: #0A1128;
  font-size: 6.4rem;
  text-transform: uppercase;
}

.c2 {
  display: grid;
  grid-template-columns: 1fr 2fr;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  gap: 3.2rem;
}

.c2 .c4 {
  margin-bottom: 4.0rem;
}

.c7 {
  max-width: 100%;
  height: auto;
}

.c1 {
  max-width: 120rem;
  margin: 0 auto;
  padding: 3.2rem;
}

.c0 {
  background: #FFFFFF;
  color: #0A1128;
  min-height: 100vh;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
}

.c6 {
  font-size: 2.4rem;
}

@media (max-width:768px) {
  .c5 {
    font-size: 4.0rem;
  }
}

@media (max-width:768px) {
  .c2 {
    grid-template-columns: 1fr;
    text-align: center;
  }
}

@media (max-width:768px) {
  .c3 {
    margin-bottom: 3.2rem;
  }
}

<div>
  <div
    class="c0"
    id=""
  >
    <div
      class="c1"
    >
      <div
        class="c2"
      >
        <div
          class="c3"
        >
          <h2
            class="c4 c5"
          >
            Grid two columns
          </h2>
          <div
            class="c6"
          >
            <p>
              Lorem ipsum dolor sit amet consectetur adipisicing elit. Distinctio magnam culpa eveniet doloribus harum? Ipsam, a necessitatibus? Sequi sunt accusantium quod, animi iure a, aliquid dolor ea vel magni dolore?
            </p>
          </div>
        </div>
        <div
          class=""
        >
          <img
            alt="Grid two columns"
            class="c7"
            src="assets/images/javascript.svg"
          />
        </div>
      </div>
    </div>
  </div>
</div>
`;
